src/types/widget.ts

```diff
@@ -37,7 +37,7 @@ type UiState = {
 export interface Widget {
   init?(options: InitOptions): void;
   render?(options: RenderOptions): void;
-  dispose?(options: DisposeOptions): SearchParameters | undefined;
+  dispose?(options: DisposeOptions): SearchParameters | void;
```

Comment:
From what I understand on [TypeScript `void` doc](https://www.typescriptlang.org/docs/handbook/basic-types.html#void), `void` accepts both `undefined` and `null`. We never return `null` in `dispose()`. This change is not a big deal though.

Reply:
With `undefined` I had this error on the `dispose` method in [`src/lib/__tests__/insights-client-test.ts`](https://github.com/algolia/instantsearch.js/blob/develop/src/lib/__tests__/insights-client-test.ts#L3-L12):

> Type '() => void' is not assignable to type '(options: DisposeOptions) => SearchParameters | undefined'.
  Type 'void' is not assignable to type 'SearchParameters | undefined'.ts(2322)

```ts
const connectHits = (renderFn: any, unmountFn: any): WidgetFactory<unknown> => (
  widgetParams = {}
): Widget => ({
  init() {},
  render({ results, instantSearchInstance }) {
    const hits = results.hits;
    renderFn({ hits, results, instantSearchInstance, widgetParams }, false);
  },
  dispose() {
    unmountFn();
  },
});
```